foam_internalwrapper/foam_internalwrapper.py

```diff
@@ -0,0 +1,260 @@
+""" foam_internalwrapper module
+
+Wrapper module for OpenFOAM
+
+"""
+from simphony.core.cuba import CUBA
+from simphony.core.data_container import DataContainer
+from simphony.cuds.abc_modeling_engine import ABCModelingEngine
+from PyFoam.Execution.ConvergenceRunner import ConvergenceRunner
+from PyFoam.LogAnalysis.BoundingLogAnalyzer import BoundingLogAnalyzer
+from foam_files import FoamFiles
+from cuba_extension import CUBAExt
+from foam_mesh import FoamMesh
+import simphonyfoaminterfaceII as foamface
+import os
+
+
+class FoamInternalWrapper(ABCModelingEngine):
+    """ Wrapper to OpenFOAM
+
+    """
+
+    def __init__(self):
+
+        self._meshes = {}
+        self.CM = DataContainer()
+        self.BC = DataContainer()
+        self.SP = DataContainer()
+        #: to be able to use CUBAExt keywords, which are not in accepted
+        #  CUBA keywords these extensions to CM and SP is used
+        self.CM_extensions = {}
+        self.SP_extensions = {}
+
+    def run(self):
+        """ run OpenFoam based on CM, BC and SP data
+
+        Returns
+        -------
+        lastTime : lastTime
```

Comment:
>> The base class description is to return anything from run method.
The last time step should be saved to mesh attribute _time.
On IO side the last time step is returned, which should also changed to saving the time to FoamMesh attribute _time. This return value is used also in examples, which should be changed. Could you do these changes as well ?

> Changes made.

I don't see these changes.  I still see the `run` returning here and also in the examples.

@khiltunen  for the file-io wrapper, the same thing applies.  maybe an issue should be created to change it at the file-io wrapper.